### ISSUE TITLE
GFA input takes a very long time

### ISSUE THREAD

**Issue by ekg** [vgteam/vg#1809]
I was trying to convert a [GFA-formatted graph produced by seqwish](http://hypervolu.me/~erik/tmp/MHC.fa.gfa.gz) to .vg format.

I used `zcat MHC.fa.gfa.gz | vg view -Fv - >MHC.fa.vg`

It took almost 10 hours. That's 4 times longer than it took to construct the graph with seqwish!

```
vg view -Fv -  25861.28s user 7.82s system 74% cpu 9:39:11.80 total                                                                                                                                                  
```

Could we implement a streaming conversion?

It's not clear to me why reading the graph in as GFA is so much more expensive than reading in a .vg format one. It would seem that we have some kind of quadratic penalty in the GFA parsing.

**Comment by edawson** [comment]
This is an issue with the paths in GFAKluge. If we restrict the same GFA to only header/segment/link lines, the same parser reads the file in 3 seconds.
```
  grep "^S\|^L\|^H" MHC.fa.gfa > hsl_lines.gfa
  time ./gfak stats hsl_lines.gfa  
  real	0m3.992s
  user	0m3.594s
  sys	0m0.353s
````

Otherwise it exceeds the length of time I'm willing to wait for it to finish. I'm hunting down where this goes wrong but I think it's in trying to harmonize the GFA1/GFA2 representations.

**Comment by edawson** [comment]
This is due to the way I did path/walk harmonization in the `paths_as_walks` and `walks_as_paths` functions. The right answer is to just rip these out and store Walks as path_elems behind the curtain. I should probably refactor this further to use group_elems for paths, and deprecate the path_elem struct entirely (or have it just shadow the group_elem struct).

The quick fix is to deprecate the `W` line type (which only vg has ever used). @ekg how do you feel about this?

**Comment by ekg** [comment]
Deprecate the W type of that's the quickest fix. No reason to use it if it
is only vg that ever implemented. Bring on the 500M strings.

**Comment by ekg** [comment]
I also don't quite understand why either P or W would cause problems in the parsing. Are we checking that the path is specified only once? Even then it should be a linear scan through the input to convert it to vg format.

**Comment by faithokamoto** [comment]
It seems this was done at the time? In any case the file is no longer accessible so I can't check. If this is still a problem then please reopen this with more details and/or open a new Issue with modern vg and a file we can use.
